casanovo/casanovo.py

```diff
@@ -382,7 +395,76 @@ def setup_model(
     return config, model
 
 
-def _get_model_weights() -> str:
+def _get_weights_from_url(
+    file_url: Optional[str],
+    cache_dir: Path,
+) -> str:
+    """
+    Attempt to download weight file from URL if weights are not already
+    cached. Otherwise use cased weights. Downloaded weight files will be
+    cached.
+
+    Parameters
+    ----------
+    file_url : str
+        url pointing to model weights file
+    cache_dir : Path
+        model weights cache directory path
+
+    Returns
+    -------
+    str
+        path to cached weights file
+    """
+    os.makedirs(cache_dir, exist_ok=True)
+    url_hash = shake_256(file_url.encode("utf-8")).hexdigest(10)
+    cache_file_name = url_hash + ".ckpt"
+    cache_file_path = cache_dir / cache_file_name
+
+    if cache_file_path.is_file():
+        logger.info(f"Model weights {file_url} retrieved from local cache")
+        return str(cache_file_path)
+
+    logger.info(f"Model weights {file_url} not in local cache, downloading")
+    file_response = requests.get(file_url)
+
+    if not file_response.ok:
+        logger.error(f"Failed to download weights from {file_url}")
+        logger.error(
+            f"Server Response: {file_response.status_code}: {file_response.reason}"
+        )
+        raise ConnectionError(f"Failed to download weights file: {file_url}")
+
+    logger.info("Model weights downloaded, writing to cache")
+    with open(cache_file_path, "wb") as cache_file:
+        cache_file.write(file_response.content)
+
+    logger.info("Model weights cached")
```

Comment:
**nitpick:** Probably not an info-level log message, It can even be omitted entirely imo.